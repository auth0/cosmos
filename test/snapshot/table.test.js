import React from 'react'
import renderer from 'react-test-renderer'
import { Table } from '@auth0/cosmos'
import 'jest-styled-components'

describe('Table', () => {
  it('renders with automation attribute', () => {
    const tree = renderer
      .create(
        <Table
          items={[
            {
              name: 'john'
            }
          ]}
        >
          <Table.Column field="name" title="Name" width="30%" />
        </Table>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
