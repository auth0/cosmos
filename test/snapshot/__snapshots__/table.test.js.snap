// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders with automation attribute 1`] = `
.c0 {
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
}

.c1 {
  padding: 8px;
  border-bottom: 2px solid #D9D9D9;
  text-align: left;
  vertical-align: bottom;
  line-height: 2;
  cursor: auto;
}

.c1:hover {
  color: inherit;
}

.c2 {
  padding-left: 8px;
  visibility: hidden;
}

.c3 {
  cursor: inherit;
}

.c3:hover {
  background: #FAFAFA;
}

.c4 {
  padding: 8px;
  border-top: 1px solid #D9D9D9;
  text-align: left;
  vertical-align: middle;
  line-height: 2;
  width: 30%;
}

<table
  className="c0"
  data-cosmos-key="table"
>
  <thead
    className=""
    data-cosmos-key="table.header"
  >
    <tr
      className=""
    >
      <th
        className="c1"
        onClick={[Function]}
      >
        Name
        <span
          className="c2"
          display={false}
        >
          â†‘
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    className=""
    data-cosmos-key="table.body"
  >
    <tr
      className="c3"
      data-cosmos-key="table.row"
      onClick={null}
    >
      <td
        className="c4"
      >
        john
      </td>
    </tr>
  </tbody>
</table>
`;
